docs/contributing.md

```diff
@@ -0,0 +1,61 @@
+# Contributing
+This section will demonstrate how to set up a remote development environment. This application utilizes the `devcontainer` directory to build a remote container that will have all necessary dependencies installed to contribute to the **Azure Key Vault Provider for Secret Store CSI Driver**.
+
+### Table of Contents
+- [Prerequisites](#prerequisites)
+- [Clone Repository](#clone-repository)
+- [Open Remote Container](#open-remote-container-in-vs-code)
+- [Debugging Introduction](#debugging-introduction)
+
+## Prerequisites
+1. Visual Studio Code with `Remote - Container` extension enabled.
+2. Azure account and subscription
+3. **optional** Azure Key Vault configured
```

Comment:
Give some thought on the path for folks who don't want to use VS Code for contributing. 

Pre-reqs could definitely be things like Go, Azure Subscription.

I would suggest considering 2 paths here:
- Local dev environment with editor/ide of choice
- Want to use VS Code with Remote Container option